pkg/validator/controllers/generic.go

```diff
@@ -0,0 +1,126 @@
+	return g.ObjectMeta
+}
+
+// GetKind returns the supportedcontroller enum type
+func (g GenericController) GetKind() config.SupportedController {
+	return g.Kind
+}
+
+// GetKindString returns a string representing what kind of object the top level controller is.
+func (g GenericController) GetKindString() string {
+	if g.KindString == "" {
+		return g.Kind.String()
+	}
+	return g.KindString
+}
+
+// GetName is inherited by all controllers using generic controller to get the name of the controller
+func (g GenericController) GetName() string {
+	return g.Name
+}
+
+// GetNamespace is inherited by all controllers using generic controller to get the namespace of the controller
+func (g GenericController) GetNamespace() string {
+	return g.Namespace
+}
+
+// LoadControllers loads a list of controllers from the kubeResources Pods
+func LoadControllers(pods []kubeAPICoreV1.Pod, dynamicClientPointer *dynamic.Interface, restMapperPointer *meta.RESTMapper) []GenericController {
+	interfaces := []GenericController{}
+	for _, pod := range pods {
+		interfaces = append(interfaces, NewGenericPodController(pod, dynamicClientPointer, restMapperPointer))
+	}
+	return deduplicateControllers(interfaces)
+}
+
+// Because the controllers with an Owner take on the name of the Owner, this eliminates any duplicates.
+// In cases like CronJobs older children can hang around, so this takes the most recent.
+func deduplicateControllers(controllers []GenericController) []GenericController {
+	controllerMap := make(map[string]GenericController)
+	for _, controller := range controllers {
+		key := controller.GetNamespace() + "/" + controller.GetKindString() + "/" + controller.Name
+		oldController, ok := controllerMap[key]
+		if !ok || controller.CreatedTime.After(oldController.CreatedTime) {
+			controllerMap[key] = controller
+		}
+	}
+	results := make([]GenericController, 0)
+	for _, controller := range controllerMap {
+		results = append(results, controller)
+	}
+	return results
+}
+
+// NewGenericPodController builds a new controller interface for anytype of Pod
+func NewGenericPodController(originalResource kubeAPICoreV1.Pod, dynamicClientPointer *dynamic.Interface, restMapperPointer *meta.RESTMapper) GenericController {
+	controller := GenericController{}
+	controller.Name = originalResource.Name
+	controller.Namespace = originalResource.Namespace
+	controller.PodSpec = originalResource.Spec
+	controller.ObjectMeta = originalResource.ObjectMeta
+	controller.Kind = config.NakedPods
+	controller.KindString = "Pod"
+	controller.CreatedTime = controller.GetObjectMeta().CreationTimestamp.Time
+
+	owners := controller.GetObjectMeta().OwnerReferences
+	if dynamicClientPointer == nil || restMapperPointer == nil {
+		return controller
+	}
+	// If an owner exists then set the name to the controller.
+	// This allows us to handle CRDs creating Controllers or DeploymentConfigs in OpenShift.
+	for len(owners) > 0 {
```

Comment:
Should we log an info message if there are multiple owners?

Reply:
Got the logging in there now.